### PR TITLE
Model Registration: Hosted Client Implementation

### PR DESCRIPTION
**Pull request by epps**
# Description

This PR introduces changes to the Python client, as well as changes to the structure of the E2E tests after discussion around the global registrar, for the Python client implementation for model registration in hosted mode.

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: client/tests/conftest.py [hunk 1]
```diff
@@ -149,24 +149,24 @@ def s3(aws_credentials):
...
-            source = provider.register_file(
-                name="transactions",
-                variant="quickstart",
-                description="A dataset of fraudulent transactions.",
-                path=f"{dir_path}/test_files/input_files/transactions.csv"
-            )
-            return (provider, source)
-        
-        return get_local
+    def get_local(is_insecure):
```
Comment (simba-git) [inline client/tests/conftest.py:152]: Is this parameter used?

File: client/src/featureform/register.py [hunk 2]
```diff
@@ -2582,7 +2582,11 @@ def get_model(self, name, local=False) -> Model:
         Returns:
             model (Model): Model
         """
-        model = get_model_info_local(name) if local else get_resource_info(self._stub, "model", name)
+        if local:
+            model = get_model_info_local(name)
+        else:
+            model_proto = get_resource_info(self._stub, "model", name)
+            model = Model(model_proto.name) if model_proto is not None else model_proto
```
Comment (simba-git) [inline client/src/featureform/register.py:2589]: Should this return the model proto or fail?

File: client/tests/conftest.py (hunk 1 above)
Comment (epps) [reply client/tests/conftest.py:152]: It's not, but given the hosted fixture requires it, I had to pass it to the local counterpart. Is there an idiomatic way of ignoring a parameter in Python?

Comment (epps) [reply client/tests/conftest.py]: It is. I needed to pass a param to the local fixture given the hosted fixture requires it. I'll convert this to an `_` to better convey the fact I don't intend to use it.

File: client/src/featureform/register.py (hunk 2 above)
Comment (epps) [reply client/src/featureform/register.py:2589]: The pattern with the other resource get methods was to print out an message to the console in the event of a gRPC error. In these cases, the implicit return type is `None` so I tried to maintain that API here.

File: client/tests/conftest.py [hunk 3]
```diff
@@ -149,24 +149,24 @@ def s3(aws_credentials):
...
-            source = provider.register_file(
-                name="transactions",
-                variant="quickstart",
-                description="A dataset of fraudulent transactions.",
-                path=f"{dir_path}/test_files/input_files/transactions.csv"
-            )
-            return (provider, source)
-        
-        return get_local
+    def get_local(_):
```
Comment (sdreyer) [inline client/tests/conftest.py:152]: Is this underscore supposed to be here?

File: client/src/featureform/register.py (hunk 2 above)
Comment (sdreyer) [reply client/src/featureform/register.py:2589]: I would just do `else None` for clarity for now. I blocked off time on my calendar this week to review that lingering PR that would remove this weird logic

File: client/tests/conftest.py (hunk 3 above)
Comment (epps) [reply client/tests/conftest.py:152]: Yes, previously, it was a named param that I don't actually need for the local fixture, but to ensure I had the same function signature for both local and hosted, I passed the param here, but changed it to an underscore to convey the fact I didn't intend to use it.

**Review by sdreyer — APPROVED** [review]

**Review by simba-git — APPROVED** [review]

**Pull request merged**